pkg/analyzer/analyzers/launchdarkly/models.go

```diff
@@ -0,0 +1,135 @@
+package launchdarkly
+
+import "sync"
+
+var (
+	MetadataKey = "key"
+
+	// resource types
+	applicationKey  = "Application"
+	repositoryKey   = "Repository"
+	projectKey      = "Project"
+	environmentKey  = "Environment"
+	experimentKey   = "Experiment"
+	holdoutsKey     = "Holdout"
+	membersKey      = "Member"
+	destinationsKey = "Destination"
+	templatesKey    = "Templates"
+	teamsKey        = "Teams"
+	webhooksKey     = "Webhooks"
+	featureFlagsKey = "Feature Flags"
+)
+
+type SecretInfo struct {
+	User        User
+	Permissions []string
+	Resources   []Resource
+	// to concurrently read and write
+	mu sync.RWMutex
+}
+
+// User is the information about the user to whom the token belongs
+type User struct {
+	AccountID string // account id. It is the owner id of token as well
+	MemberID  string
+	Name      string
+	Role      string // role of caller
+	Email     string
+	Token     Token
+}
+
+// Token is the token details
+type Token struct {
+	ID             string       // id of the token
+	Name           string       // name of the token
+	CustomRoles    []CustomRole // custom roles assigned to the token
+	InlineRole     []Policy     // any policy statements maybe used in place of a built-in custom role
+	Role           string       // role of token
+	IsServiceToken bool         // is a service token or not
+	APIVersion     int          // default api version assigned to the token
+}
+
+// CustomRole is a flexible policies providing fine-grained access control to everything in launch darkly
+type CustomRole struct {
+	ID                string
+	Key               string
+	Name              string
+	Polices           []Policy
+	BasePermission    string
+	AssignedToMembers int
+	AssignedToTeams   int
+}
+
+/*
+policy is a set of statements
+
+Jargon:
+  - Resource: List of resources
+  - NotResources: Except this list of resources
+  - Actions: List of actions
+  - NotActions: Except this list of actions
+  - Effect: Allowed or Denied
+*/
+type Policy struct {
+	Resources    []string
+	NotResources []string
+	Actions      []string
+	NotActions   []string
+	Effect       string
+}
+
+type Resource struct {
+	ID             string
+	Name           string
+	Permission     string
+	Type           string
+	ParentResource *Resource
+	MetaData       map[string]string
+}
+
+// appendResources append resource to secret info resources list
+func (s *SecretInfo) appendResource(resource Resource) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	s.Resources = append(s.Resources, resource)
+}
+
+// listResourceByType returns a list of resources matching the given type.
+func (s *SecretInfo) listResourceByType(resourceType string) []Resource {
+	s.mu.RLock()
```

Comment:
question: Do we actually need the mutex? 🤔  I didn’t notice any concurrent operations, but I might have missed something.

Reply:
I guess you missed the `requests.go` file.

Reply:
I guess you missed the `requests.go` file. Large diffs are not rendered by default.

Reply:
😢 I knew there had to be a reason. Sorry about that.

Reply:
No worries!